File: easy/onboarding/Haskell/onboarding.hs
import System.IO
import Control.Monad

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering -- DO NOT REMOVE
    
    -- CodinGame planet is being attacked by slimy insectoid aliens.
    -- <---
    -- Hint: To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
    
    -- game loop
    forever $ do
        enemy1 <- getLine -- name of enemy 1
        dist1 <- readLn :: IO Int -- distance to enemy 1
        enemy2 <- getLine -- name of enemy 2
        dist2 <- readLn :: IO Int -- distance to enemy 2
        
        -- hPutStrLn stderr "Debug messages..."
        
        -- Determine which enemy is closer and print its name
        putStrLn $ if dist1 < dist2 then enemy1 else enemy2

File: easy/onboarding/PHP/onboarding.php
<?php
// game loop
while (TRUE)
{
    fscanf(STDIN, "%s", $enemy1); // name of enemy 1
    fscanf(STDIN, "%d", $dist1); // distance to enemy 1
    fscanf(STDIN, "%s", $enemy2); // name of enemy 2
    fscanf(STDIN, "%d", $dist2); // distance to enemy 2

    // Write an action using echo(). DON'T FORGET THE TRAILING \n

    // Determine which enemy is closer and print its name
    echo ($dist1 < $dist2 ? $enemy1 : $enemy2) . "\n";
}
